src/CoreBundle/Resources/public/javascript/table.js

```diff
@@ -181,52 +181,96 @@ $(document).ready(function () {
-        // select2 does not expose any way to retrieve it. Therefore the DOM will be queried for the
-        // search field's selector.
-        // The query will return the first search field found in the DOM which might not be the desired one.
-        // Yet as select2 creates a new search field on activating the widget and removes it when unnecessary there
-        // should be only one search field present in the DOM.
+    }).on('select2:closing', function(e) {
+         /* There is no handler attached to the `select` event. Instead the `closing` handler will also manage the
+         * `select` event. This is done by analyzing the data passed alongside the event.
+         * */
+
+        const originalSelect2EventData = e.params.args.originalSelect2Event;
+        if ('undefined' === typeof originalSelect2EventData) {
+            return;
+        }
+
+        const data = originalSelect2EventData.data;
+
+        if ('undefined' === typeof data) {
+            return;
+        }
+
+        /* The following condition validates to `true` when no selection has been made from select2's suggestions.
+        *  In this case a hidden form field with the search term as value will be added to the form.
+        * */
+        if (true === data.newOption) {
+            if ('' === data.text) {
+                return;
+            }
+
+            const form = createSearchWordInputWithValue(data.text.trim());
 
+            form.submit();
+        } else {
+            switchRecord(data.id.trim(), data.text.trim());
+        }
+    }).on('select2:open', function(e) {
+        /* Synchronize select2's search term display field by listening to keyboard events. */
         const select2SearchField = document.querySelector('.select2-search__field');
+        const select2 = getSibling(this, '.select2');
+        const selectSelectionRendered = null === select2 ? null : select2.querySelector('.select2-selection__rendered');
+
+        if (null === select2SearchField || null === selectSelectionRendered) {
+            return;
+        }
+
+        const text = $(selectSelectionRendered).contents().filter(function() {
+            return this.nodeType === Node.TEXT_NODE;
+        }).text();
+
+        select2SearchField.value = text;
+        select2SearchField.select();
+
+        $(select2SearchField).off('input');
 
-        if (null === select2SearchField || '' === select2SearchField.value ) {
+        if ('1' === this.dataset.keyUpListenerAttached) {
             return;
         }
+        select2SearchField.addEventListener('keyup', (e) => {
+            selectSelectionRendered.title = select2SearchField.value;
+            selectSelectionRendered.textContent = select2SearchField.value;
+        });
 
-        createSearchWordInputWithValue(select2SearchField.value);
+        this.dataset.keyUpListenerAttached = '1';
     });
 
-    function switchRecord(id) {
+    function getSibling(node, selector) {
```

Comment:
Doesn't jquery do this with sibling() or similar?

Reply:
It does. Thanks.